### PR TITLE
Stage Reporting And Availability Metric

### PR DESCRIPTION
**Pull request by Boomatang**
ISSUE: https://issues.redhat.com/browse/INTLY-5312

# What
Add top level stage indicator to the status block in the RHMI CR. This stage indicator shows the current stage of installation and is marked as complete when all stages have been completed.

Add a metric which indicates the availability of the install. 

# Verify
## Stage Indicator
* Start the operator install process
* Watch the RHMI CR for updates.
* During the installation, the `status.stage` should display the current stage been installed.
* Once install is complete `status.stage` == `complete`

## Metric update
* polling the metric url `curl <cluster>:8383/metrics | grep rhmi` the `rhmi_status_available` metric will be displayed.
* `rhmi_status_available` will state `0` while the installation process continues
* `rhmi_status_available` should `1` once the install is complete
* During an uninstall this metric should be set to `0`

### CODE REVIEW

File: pkg/apis/integreatly/v1alpha1/rhmi_types.go [hunk 1]
```diff
@@ -133,6 +133,7 @@ type RHMIStatus struct {
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
 	Stages             map[StageName]RHMIStageStatus `json:"stages"`
+	Stage			   StageName 					 `json:"stage"`
```
[the remark this answers is not in the document]
Comment (Boomatang) [reply pkg/apis/integreatly/v1alpha1/rhmi_types.go]: Ok I have done that and force push the changes. Was there anything else that I should have ran before doing the PR?

**Comment by philbrookes** [comment]
bah, sorry, closed this by mistake!

**Pull request merged**